src/project-manager.ts

```diff
@@ -872,9 +872,14 @@ export class ProjectConfiguration {
 	private updateDiagnostics(program: ts.Program, childOf = new Span()): void {
 		const span = childOf.tracer().startSpan('Update diagnostics', { childOf });
 		try {
-			const diagnostics = ts.getPreEmitDiagnostics(program);
-			span.log({ event: 'result', result: diagnostics.length });
+			let diagnostics: ts.Diagnostic[] = [];
+			for (const file of program.getSourceFiles()) {
+				if (!file.fileName.includes('/node_modules/') && util.isJSTSFile(file.fileName)) {
```

Comment:
does this work on Windows? I would call `toUnixPath()` before or use `/[\/\\]node_modules[\/\\]/.test()`

why the check for `.isJSTSFile()`? What other files could there be, a tsconfig.json?